### ISSUE TITLE
Checkout with iDeal, Bancontact and giropay fails randomly when using the Checkout block

### ISSUE THREAD

**Issue by paulostp** [Automattic/woocommerce-payments#5386]
### Describe the bug

For some merchants from the Netherlands, using the iDeal payment method, sometimes the checkout will fail with the error: `"Invalid paramater owner [name]. Owner name must be at least 3 characters long"`.

![image](https://user-images.githubusercontent.com/69197772/211784390-7a4b1322-8953-4c01-87c0-815a740f62a3.png)

* The Stripe logs show that a request to confirm a payment intent failed because of `Invalid paramater owner [name]`. Inspecting the log further, shows that **all the customer data is missing from the request**:

![image](https://user-images.githubusercontent.com/69197772/211785828-39fa1a31-0f90-4782-ae7d-0a8b26a0b752.png)

* **This is a random error and the checkout _can_ be successful if you try again.**

![image](https://user-images.githubusercontent.com/69197772/211786764-320ba3d0-d761-46cb-8366-052e28975eba.png)

### To Reproduce

**Update**: [steps to reproduce](https://github.com/Automattic/woocommerce-payments/issues/5386#issuecomment-1590555123)

It seems to be caused by the Checkout block. At some point, we were able to reproduce the behavior on a test site by switching the checkout page from shortcode to block.

![image](https://user-images.githubusercontent.com/69197772/211784727-40480da3-fae8-405b-beee-bb25ca77dfa6.png)

However, due to the random nature of the issue, haven't had success in reproducing it further. **The merchant does mention this is more frequent with mobile devices**.

### Actual behavior
The checkout fails, hinting that some information is invalid, even though the the whole checkout form is filled.

### Expected behavior
That the merchant gets successfully redirected to iDeal if the checkout form has all the necessary information

### Additional context

The latest case is 5802398-zen.
Other earlier instances of this same error:
* 5649128-zen (merchant went for a conflict test and never came back, no transactions in WooCommerce Payments since)
* 5689828-zen (merchant solved the issue by tweaking their checkout page)
* 5747899-zen (merchant reported that they identified that the Checkout block is causing this issue)
* 5683211-zen (merchant made some changes to their site and the error disappeared, no transactions in WooCommerce Payments since)

**Comment by jessy-p** [comment]
@Automattic/pulsar

**Comment by kalessil** [comment]
Thanks for the ping @jessy-p! We'll look into details in ~2 weeks - currently, we don't have the needed capacity.

**Comment by kaushikasomaiya** [comment]
6041100-zen

**Comment by kalessil** [comment]
👀

**Comment by eriklenaerts** [comment]
hey, I have the same issue here with the `Bankcontact` payment option. I think the error occurs whenever I skipped the billing details and select that payment option first.

**Comment by maxlaf** [comment]
6265024-zen

**Comment by senff** [comment]
6271250-zen

**Comment by nagpai** [comment]
I did a test today on my local setup with a Netherlands Account and with iDEAL . I was unable to replicate the error. 

<img width="905" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/4162931/087cf57f-f16c-4be3-a503-eb4563c54a4e">

![image](https://github.com/Automattic/woocommerce-payments/assets/4162931/e5eb5368-e2df-4316-aed2-bf385f380f36)

It could be a plugin conflict that prevents the fields from being submitted. 

Can we ask the users to test this with Storefront / TwentyTwenty three and with all plugins turned off except Woo + WCPay ?

**Comment by paulostp** [comment]
> It could be a plugin conflict that prevents the fields from being submitted. Can we ask the users to test this with Storefront / TwentyTwenty three and with all plugins turned off except Woo + WCPay ?

A conflict is always a possibility, but all of these sites are very different. Back then I was able to reproduce once on my test site with Storefront. Due to the random nature of this misbehavior, I wasn't able to see it again, just like the original merchant reported, if you try again it will usually work.

This isn't something that we can ask merchants to experiment with on a live store as most times the merchant's themselves can't reproduce — it's the customers that report it. It would be a different scenario if there was no workaround, but we already gave the merchants a solution by asking them to use the shortcode instead of the checkout block. Their payment problems are solved for now, even if the checkout page isn't looking like they wanted to. 

However, the solution for some of these merchants was to stop using WooCommerce Payments, which is a shame.

**Comment by nagpai** [comment]
>Back then I was able to reproduce once on my test site with Storefront

Cool. If we are able to reproduce this on a test setup then that proves that the issue is not related to other plugins / theme. 
As you rightly pointed out, @paulostp , the issue is pretty random and difficult to reproduce. Hence it is difficult to find out accurately what could be the cause of it. 

<s>From the screenshots and nature of the issue, it seems the information is blocked at the browser side itself. For future instances, if we are able to reproduce, or if we have contact with the user that is facing the issue, can we get the browser details through a site like [whatismybrowser.com ](https://www.whatismybrowser.com)</s>

**Comment by nagpai** [comment]
Thank you, @eriklenaerts  and sorry about missing your comment earlier. 

I was able to consistently reproduce the bug with these steps:

1. Open the website in incognito  and check out as a guest user, through a checkout pages made using Blocks. 
2. Make sure the billing fields, except the prefilled country field, are all empty
3. Select the payment Method ( iDEAL - also reproduced when we try Bancontact ) 
4. Try to checkout and you will see the error. `Invalid parameter owner[name]. Owner name must be at least 3 characters long.`
5. Payment Intent confirmation API call shows a payload with all billing fields empty, except country

<img width="751" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/4162931/074f47f4-0967-4f13-8c5f-bc0d29633c65">

**Comment by eriklenaerts** [comment]
@nagpai that's good news ;). thanks for looking into it

**Comment by rossviviano** [comment]
6751972-zen

**Comment by timur27** [comment]
I attempted to replicate the issue in two different environments, but I was unsuccessful in both

1. [JN site with WooPayments `5.3.3`](https://futuristically-iron.jurassic.ninja/). I use this WooPayments version to be sure that the environment runs legacy UPE (this issue describes legacy UPE fields) and is more in line with the timeline the issue was reported. My site is running the latest version of WooCommerce -> Blocks, and I guess that checkout fields are more bulletproof if it goes to validating empty fields. [I'm seeing a lot of field validation errors](https://github.com/Automattic/woocommerce-payments/assets/22319444/1233dfe8-825d-447f-a440-eee85d11a47b) as expected, when trying to reproduce.
2. Local environment with deferred intent creation UPE. This is a different type of checkout implementation which performs steps differently and overall, a lot has been changed there compared to the legacy UPE. This could also lead to the point that the issue is not reproducible now. 

To sum up, the issue is not reproducible now and I'm closing it. I will follow if there will be new reports and we will re-open it then.
